ansible/roles/ci/code_style_check/catkin/files/catkin_lint_xunit.py

```diff
@@ -0,0 +1,189 @@
+        if args.lintignore:
+            lintignore_path = packagename+'/'+args.lintignore
+            if os.path.isfile(lintignore_path):
+                if os.stat(lintignore_path).st_size == 0:
+                    # ignore this package
+                    skip_package=True
+                else:
+                    with open(lintignore_path) as f:
+                        lintignore_lines = f.read().splitlines()
+                    full_ignore=','.join(lintignore_lines)
+                    cmd = [catkinlint_bin, '-W0','-q','--ignore',full_ignore, packagename]
+        if skip_package:
+            continue
+
+        try:
+            subprocess.check_output(
+                cmd, cwd=os.path.dirname(packagename), stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            errors = e.output.decode()
+        else:
+            errors = None
+
+        report.append((packagename, errors))
+
+    for (packagename, errors) in report:
+        if errors is not None:
+            print("Package '%s' is invalid:" % packagename, file=sys.stderr)
+            for line in errors.splitlines():
+                print(os.path.dirname(packagename)+'/'+line, file=sys.stderr)
+            print('', file=sys.stderr)
+        else:
+            print("Package '%s' is valid" % packagename)
+            print('')
+
+    # output summary
+    error_count = sum(1 if r[1] else 0 for r in report)
+    if not error_count:
+        print('No problems found')
+        rc = 0
+    else:
+        print('%d package(s) are invalid' % error_count, file=sys.stderr)
+        rc = 1
+
+    # generate xunit file
+    if args.xunit_file:
+        folder_name = os.path.basename(os.path.dirname(args.xunit_file))
+        file_name = os.path.basename(args.xunit_file)
+        suffix = '.xml'
+        if file_name.endswith(suffix):
+            file_name = file_name[0:-len(suffix)]
+            suffix = '.xunit'
+            if file_name.endswith(suffix):
+                file_name = file_name[0:-len(suffix)]
+        testname = '%s.%s' % (folder_name, file_name)
+
+        xml = get_xunit_content(report, testname, time.time() - start_time)
+        path = os.path.dirname(os.path.abspath(args.xunit_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    return rc
+
+
+def get_xunit_content(report, testname, elapsed):
+    test_count = len(report)
+    error_count = len([r for r in report if r[1]])
+    data = {
+        'testname': testname,
+        'test_count': test_count,
+        'error_count': error_count,
+        'time': '%.3f' % round(elapsed, 3),
+    }
+    xml = """<?xml version="1.0" encoding="UTF-8"?>
+<testsuite
+  name="%(testname)s"
+  tests="%(test_count)d"
+  errors="0"
+  failures="%(error_count)d"
+  time="%(time)s"
+>
+""" % data
+
+    for (packagename, diff_lines) in report:
```

Comment:
what is diff_lines? the error?

Reply:
Changed it to just: lines, there are lines in the error report